jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerCredentialRetrieverFactory.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.registry.credentials;
+
+/** Factory class for constructing {@link DockerCredentialRetriever}. */
+public class DockerCredentialRetrieverFactory {
+
+  private final String registry;
+
+  public DockerCredentialRetrieverFactory(String registry) {
+    this.registry = registry;
+  }
+
+  public DockerCredentialRetriever withSuffix(String credentialHelperSuffix) {
```

Comment:
Nit: maybe this should be called `withCredentialHelperSuffix`? or the class called `DockerCredentialHelperRetreiverFactory`?

Reply:
I'll rename `DockerCredentialRetriever` to `DockerCredentialHelper` and refactor this factory and method naming in a followup refactoring CL.